geomesa-utils/src/main/scala/org/locationtech/geomesa/utils/stats/StatSerialization.scala

```diff
@@ -0,0 +1,265 @@
+        }
+        rh
+      case _ if attrType == classOf[java.lang.Long] =>
+        val rh = new RangeHistogram[java.lang.Long](attrIndex, attrTypeString, numBins.toInt, lowerEndpoint.toLong, upperEndpoint.toLong)
+        keyValues.foreach {
+          case (keyValuePair) =>
+            val splitKeyValuePair = keyValuePair.split("->")
+            rh.histogram.put(splitKeyValuePair(0).toLong, splitKeyValuePair(1).toLong)
+        }
+        rh
+      case _ if attrType == classOf[java.lang.Float] =>
+        val rh = new RangeHistogram[java.lang.Float](attrIndex, attrTypeString, numBins.toInt, lowerEndpoint.toFloat, upperEndpoint.toFloat)
+        keyValues.foreach {
+          case (keyValuePair) =>
+            val splitKeyValuePair = keyValuePair.split("->")
+            rh.histogram.put(splitKeyValuePair(0).toFloat, splitKeyValuePair(1).toLong)
+        }
+        rh
+      case _ if attrType == classOf[java.lang.Double] =>
+        val rh = new RangeHistogram[java.lang.Double](attrIndex, attrTypeString, numBins.toInt, lowerEndpoint.toDouble, upperEndpoint.toDouble)
+        keyValues.foreach {
+          case (keyValuePair) =>
+            val splitKeyValuePair = keyValuePair.split("->")
+            rh.histogram.put(splitKeyValuePair(0).toDouble, splitKeyValuePair(1).toLong)
+        }
+        rh
+    }
+  }
+
+  /**
+   * Uses individual stat pack methods to serialize the stat
+   *
+   * @param stat the given stat to serialize
+   * @return serialized stat
+   */
+  def pack(stat: Stat): Array[Byte] = {
+    stat match {
+      case mm: MinMax[_]                => packMinMax(mm)
+      case isc: IteratorStackCounter    => packISC(isc)
+      case eh: EnumeratedHistogram[_]   => packEnumeratedHistogram(eh)
+      case rh: RangeHistogram[_]        => packRangeHistogram(rh)
+      case seq: SeqStat                 => Bytes.concat(seq.stats.map(pack) : _*)
+    }
+  }
+
+  /**
+   * Deserializes the stat
+   *
+   * @param bytes the serialized stat
+   * @return deserialized stat
+   */
+  def unpack(bytes: Array[Byte]): Stat = {
+    val returnStats: ArrayBuffer[Stat] = new mutable.ArrayBuffer[Stat]()
+    val bb = ByteBuffer.wrap(bytes)
+
+    var bytePointer = 0
+    while (bytePointer < bytes.length - 1) {
+      val statType = bytes(bytePointer)
+      val statSize = bb.getInt(bytePointer + 1)
+      val statBytes = bytes.slice(bytePointer + 5, bytePointer + 5 + statSize)
+
+      statType match {
+        case MINMAX_BYTE =>
+          val stat = unpackMinMax(statBytes)
+          returnStats += stat
+        case ISC_BYTE =>
+          val stat = unpackIteratorStackCounter(statBytes)
+          returnStats += stat
+        case EH_BYTE =>
+          val stat = unpackEnumeratedHistogram(statBytes)
+          returnStats += stat
+        case RH_BYTE =>
+          val stat = unpackRangeHistogram(statBytes)
+          returnStats += stat
+      }
+
+      bytePointer += statSize + 5
```

Comment:
[style] might be a little cleaner to use the native byte buffer position, length, etc methods and avoid manually tracking the byte pointer